### DISCUSSION TITLE
Collection level Provider

### DISCUSSION THREAD

**Discussion by rpfaeffle** [payloadcms/payload#1058] [Feature Requests & Ideas]
I'm currently working on a custom plugin field which relies on the `useField` hook and data generated by other component based fields. For a sake of simplicity I would like to create a provider which allows all custom views to post data.

With the current approach of global providers this is not possible due to the fact that payload inserts the global providers before the `<Form />` provider. Therefore I would like to ask whether its possible to add collection based providers which would be able to access the `useField` hook.

Thanks in advance

**Comment by jmikrut** [comment]
Hey @KaterBasilisk6 — interesting question. Follow-up for you.

It sounds like this is _**only applicable to your custom added views**_. Can you not build out a custom view that includes your own provider, within your form?

What do your custom views look like? Are you re-using the built-in Payload collection `Edit` view?

**Reply by rpfaeffle** [reply]
Hey @jmikrut - thanks for the quick reply.

I'm currently working on a timeline component which relies on the input of different custom components / fields. One option would definitely be to build a completely custom view. Currently everything relies on the built-in Payload collection `Edit` view and **custom fields**.
